src/storage/src/storage_state.rs

```diff
@@ -976,16 +976,51 @@ impl<'w, A: Allocate> Worker<'w, A> {
                 }
                 StorageCommand::RunIngestions(ingestions) => {
                     ingestions.retain_mut(|ingestion| {
+                        // Subsources can be dropped independently of their
+                        // primary source, so we evaluate them in a separate
+                        // loop.
+                        for export_id in ingestion
+                            .description
+                            .source_exports
+                            .keys()
+                            .filter(|export_id| **export_id != ingestion.id)
+                        {
+                            if drop_commands.remove(export_id)
+                                || self.storage_state.dropped_ids.contains(&ingestion.id)
```

Comment:
```suggestion
```
Nit: this extra condition isn't necessary since it's being checked right below too, and the subsources/source_exports will be dropped then.

It's not a big deal to leave it here but it adds to the mental overhead of reading this logic

Reply:
Excellent, will remove that!